cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -309,15 +309,15 @@ func (c *Client) GetClusterVersions() (map[string]string, error) {
 
 // ClusterAvailable returns true if the cluster status indicates the cluster is available.
 func (c *Client) ClusterAvailable() (bool, error) {
```

Comment:
AFAIK this function is used only by etcd.go, so IMO this method can go away from the interface...
Opinions?

Reply:
It is used in the following places:
```
cmd/kubeadm/app/phases/upgrade/compute_test.go:func (f fakeEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
cmd/kubeadm/app/phases/upgrade/staticpods_test.go:func (c fakeTLSEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
cmd/kubeadm/app/phases/upgrade/staticpods_test.go:func (c fakePodManifestEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
```

Reply:
This confirms the fund isn't used in any logic, but  only in test when implementing fake objects,
Do you see some problems in removing this function from the interface/from fake objects?